shell/components/form/GithubPicker.vue

```diff
@@ -1,343 +1,355 @@
 
-  async mounted() {
-    await this.fetchRepos();
+        this.branches = res;
+      } finally {
+        this.loadingBranches = false;
+      }
+    },
+    async fetchCommits() {
+      this.loadingCommits = true;
+      this.showSelections = false;
+      this.selectedCommit = false;
 
-    if ( this.value && this.value.repo ) {
-      const repo = await this.$store.dispatch('github/fetchRepoByUrl', this.value.repo);
+      try {
+        const res = await this.$store.dispatch('github/fetchCommits', {
+          repo:     this.selectedRepo,
+          username: this.selectedUsername,
+          branch:   this.selectedBranch,
+        });
 
-      if ( !repo ) {
+        this.commits = res;
+      } finally {
+        this.loadingCommits = false;
+      }
+    },
+    prepareArray(arr, commits) {
+      if (!isArray(arr)) {
         return;
       }
 
-      this.selectRepo(repo);
-
-      const branch = await this.$store.dispatch('github/fetchBranch', { repo, branch: this.value.branch });
+      if (commits) {
+        // Simplify the commits structure in the array
+
+        return arr.reduce((acc, cur) => {
+          acc.push({
+            message:   cur.commit.message,
+            html_url:  cur.html_url,
+            sha:       this.trimCommit(cur.sha),
+            commitId:  cur?.sha,
+            author:    cur.author,
+            isChecked: false,
+          });
 
-      if ( !branch ) {
-        return;
+          return acc;
+        }, []);
       }
 
-      this.selectBranch(branch);
+      // Handles repos & branches
+      const res = arr.reduce((acc, cur) => {
+        acc.push(cur.name);
 
-      const file = await this.$store.dispatch('github/fetchFile', {
-        repo,
-        branch,
-        file: this.value.file || this.preferredFile
-      });
+        return acc;
+      }, []);
 
-      if ( !file ) {
+      return res;
+    },
+    trimCommit(commit) {
+      if (!commit) {
         return;
       }
 
-      this.selectFile(file);
-    }
-  },
-
-  methods: {
-    update() {
-      if ( this.selectedRepo && this.selectedBranch && this.selectedFile ) {
-        // Do something
-      }
+      return commit.slice(0, 7);
     },
+    final(commitId) {
+      this.selectedCommit = this.commits.filter(ele => ele.sha === commitId)[0];
 
-    selectRepo(repo) {
-      this.selectedFile = null;
-      this.selectedBranch = null;
-      this.selectedRepo = repo;
-      if ( repo ) {
-        this.fetchBranches(repo);
-      }
-
-      this.update();
-    },
+      if (this.selectedUsername && this.selectedRepo && this.selectedCommit) {
+        const url = `https://github.com/${ this.selectedUsername }/${ this.selectedRepo }`;
 
-    selectBranch(branch) {
-      this.selectedFile = null;
-      this.selectedBranch = branch;
-      if ( branch ) {
-        this.fetchFiles(this.selectedRepo, branch);
+        this.$emit('generateUrl', url, this.selectedUsername, this.selectedCommit.sha);
```

Comment:
Minor one on terminology, we're emitting values in this component and it's up to the parent component what happens to them (in this case generating a url). For this case would recommend using [v-model](https://v2.vuejs.org/v2/guide/components.html#Using-v-model-on-Components) instead of a `generateUrl` property